src/mongocxx/test_util/client_helpers.cpp

```diff
@@ -343,7 +343,7 @@ bool matches(types::bson_value::view main,
                 continue;
             }
             if (main_view.find(el.key()) == main_view.end()) {
-                return false;
+                return el.get_value().type() == type::k_null;
```

Comment:
See [this](https://github.com/mongodb/specifications/commit/92255e0e57027646e8ef23399d3f33fdc114221f#diff-c4a2b694daca296938e31db2cd9b5206R289-R296)